rmw_zenoh_cpp/src/detail/rmw_data_types.cpp

```diff
@@ -70,3 +77,89 @@ void sub_data_handler(
 
   z_drop(z_move(keystr));
 }
+
+
+//==============================================================================
+std::size_t rmw_service_data_t::get_new_uid()
+{
+  return client_count++;
+}
+
+//==============================================================================
+void service_data_handler(const z_query_t * query, void * data)
+{
+  RCUTILS_LOG_INFO_NAMED(
+    "rmw_zenoh_cpp",
+    "[service_data_handler] triggered"
+  );
+  z_owned_str_t keystr = z_keyexpr_to_string(z_query_keyexpr(query));
+
```

Comment:
I'm going to suggest that we use an` rcpputils::scope_exit` to drop the `keystr`.  That way, we can remove the explicit `z_drop` from all code paths and just rely on the scope_exit (which is how we've generally done it elsewhere in `rmw_zenoh`).

Reply:
I ended up fixing this in [177451a](https://github.com/ros2/rmw_zenoh/pull/86/commits/177451a0a200e3327afe9029b5cc405731f527f5).